qiskit/transpiler/passes/calibration/rzx_builder.py

```diff
@@ -308,21 +293,12 @@ def get_calibration(self, node_op: CircuitInst, qubits: List) -> Union[Schedule,
 
         # Determine native direction, assuming only single drive channel per qubit.
         # This guarantees channel and qubit index equality.
-        is_native = comp_tones[0].channel.index == qubits[1]
-
-        stretched_cr_tone = self.rescale_cr_inst(cr_tones[0], 2 * theta)
-        stretched_comp_tone = self.rescale_cr_inst(comp_tones[0], 2 * theta)
-
-        if is_native:
-            # Placeholder to make pulse gate work
-            delay = Delay(stretched_cr_tone.duration, DriveChannel(qubits[0]))
-
-            # This doesn't remove unwanted instruction such as ZI
-            # These terms are eliminated along with other gates around the pulse gate.
-            rzx_theta = rzx_theta.insert(0, stretched_cr_tone, inplace=True)
-            rzx_theta = rzx_theta.insert(0, stretched_comp_tone, inplace=True)
-            rzx_theta = rzx_theta.insert(0, delay, inplace=True)
-
+        if comp_tones[0].channel.index == qubits[1]:
+            with builder.build(default_alignment="left", name="rzx(%.3f)" % theta) as rzx_theta:
+                stretched_dur = self.rescale_cr_inst(cr_tones[0], 2 * theta)
+                self.rescale_cr_inst(comp_tones[0], 2 * theta)
+                # Placeholder to make pulse gate work
+                builder.delay(stretched_dur, DriveChannel(qubits[0]))
```

Comment:
Previously pulse gate only consisting of control channels caused 9999 error (and we needed at least one drive channel) in execution with IBM backends. I think this bug has already been fixed, but here I just don't remove this because removal of this is not the scope of the PR.

Reply:
Ah right I remember that. Thanks for keeping it in.